src/cpu/x64/jit_generator.hpp

```diff
@@ -673,6 +673,16 @@ class jit_generator : public Xbyak::CodeGenerator, public c_compatible {
         }
     }
 
+    void uni_vhsubps(const Xbyak::Xmm &x, const Xbyak::Xmm &x2,
+                     const Xbyak::Operand &op) {
+        if (is_valid_isa(avx)) {
+            vhsubps(x, x2, op);
+        } else {
+            assert(x.isEqualIfNotInherited(op));
```

Comment:
Should it be assert(x.isEqualIfNotInherited(**x2**)); ?

Reply:
I replaced the construction with if (<condition>) { movups(x, x2); } - it should just work.
Since it was copy-pasted from uni_vhaddps() I changed it in both places.